sycl/doc/extensions/supported/sycl_ext_intel_queue_immediate_command_list.asciidoc

```diff
@@ -0,0 +1,160 @@
+for other Intel GPUs.
+
+
+== Backend support status
+
+The properties added by this extension are a hint, which all backends accept.
+However, in the current {dpcpp} implementation, the hint is only meaningful
+on the Level Zero backend.
+
+== Overview
+
+This extension enables specifying one of two submission modes for a queue.
+
+The property `no_immediate_command_list` specifies that submissions to a SYCL
+queue should be collected into groups before actual submission to the hardware.
+
+The property `immediate_command_list` specifies that submissions to a
+SYCL queue should be submitted to the hardware immediately.
+
+When the submission mode is left unspecified a platform-specific default
+is chosen by the implementation.
+
+When commands are submitted to a SYCL queue that uses the Level Zero backend,
+those commands can be submitted to the hardware in one of two ways:
+either through an immediate command list or through a standard command queue.
+Commands submitted through an immediate command list are immediately submitted
+to the device while commands submitted through a standard command queue may be
+batched with other commands before they are submitted. By default the
+implementation chooses a method that works best for most workloads.
+
+In most cases, applications should rely on the default behavior.
+However, sometimes it is advantageous for the application to choose one method
+or the other. This extension provides a way for applications to select either
+of these two methods via a queue property.
+
+For example, when kernel runtimes are very short, the submission time on the
+host may be as long or longer than the actual runtime of the kernel. In this
+case, doing batched submissions may be preferable so that the submission
+overhead is amortized over a number of kernel executions.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_QUEUE_IMMEDIATE_COMMAND_LIST` to one of the values
+defined in the table below.  Applications can test for the existence of this
+macro to determine if the implementation supports this feature, or
+applications can test the macro's value to determine which of the
+extension's features the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+=== New queue properties
+This extension adds the following new properties that can be used when
+constructing a queue object.
+
+```c++
+namespace ext {
+namespace intel {
+
+// Use immediate command lists
+property_list P1{property::queue::immediate_command_list()};
+queue Q1{P1};       
+
+// Do not use immediate command lists
+property_list P2{property::queue::no_immediate_command_list()};
+queue Q2{P2};
+
+...
+
+} // namespace intel
+} // namespace ext
+```
```

Comment:
A synopsis should show the declarations of the properties, not an example usage.

```
namespace sycl::ext::intel::property::queue {

struct immediate_command_list {};
struct no_immediate_command_list {};

} // namespace sycl::ext::intel::property::queue
```